src/fv3jeditools/diag_field_plot.py

```diff
@@ -67,86 +79,88 @@ def field_plot(datetime, conf):
+          outfile = fields_file_name+"_"+field_name+"_layer-"+str(model_layer)+".png"
 
-    # Check if field is two or three dimensions
-    if len(ncfile.variables[field_name].shape) == 4:
+        elif len(ncfile.variables[field_name].shape) == 3:
 
-      # User must provide layer/level to plot if 3D
-      if (model_layer == None):
-          utils.abort("If plotting 3D variable user must provide \'model layer\' in the configuration")
+          # Message and read the field at provided layer
+          print(" Reading field ", field_name)
+          field[:,:] = ncfile.variables[field_name][:,:]
+          title = "Contour of "+field_name+" ("+units+")"
+          outfile = fields_file_name+"_"+field_name+".png"
 
-      # Message and read the field at provided layer
-      print(" Reading layer ", model_layer, " from field ", field_name)
-      field[:,:] = ncfile.variables[field_name][:,model_layer-1,:,:]
+        # Check if field has positve and negative values
+        # ----------------------------------------------
+        if np.min(field) < 0:
+          cmax = np.max(np.abs(field))
+          cmin = -cmax
+          cmap = 'RdBu'
+        else:
+          cmax = np.max(field)
+          cmin = np.min(field)
+          cmap = 'nipy_spectral'
 
-      # Set plot title and output file to include level plotted
-      title = "Contour of "+field_name+" ("+units+") for layer "+str(model_layer)
-      outfile = os.path.splitext(fields_file)[0]+"_"+field_name+"_layer-"+str(model_layer)+".png"
+        levels = np.linspace(cmin,cmax,25)
 
-    elif len(ncfile.variables[field_name].shape) == 3:
+        # Create two dimensional contour plot of field
+        # --------------------------------------------
 
-      # Message and read the field at provided layer
-      print(" Reading field ", field_name)
-      field[:,:] = ncfile.variables[field_name][:,:]
-      title = "Contour of "+field_name+" ("+units+")"
-      outfile = os.path.splitext(fields_file)[0]+"_"+field_name+".png"
+        # Set the projection
+        projection = ccrs.PlateCarree()
+
+        # Create figure to hold plot
+        fig = plt.figure(figsize=(10, 5))
+
+        # Just one subplot for now
+        ax = fig.add_subplot(1, 1, 1, projection=projection)
+
+        # Contour the field
+        im = ax.contourf(lons, lats, field,
+                         transform=projection,
+                         cmap=cmap,
+                         levels=levels)
+
+        # Add coast lines to the plot
+        ax.coastlines()
+
+        # Add labels to the plot
+        ax.set_xticks(np.linspace(-180, 180, 5), crs=projection)
+        ax.set_yticks(np.linspace(-90, 90, 5), crs=projection)
+        ax.set_xlabel('Longitude')
+        ax.set_ylabel('Latitude')
+        ax.set_title(title)
+        ax.set_global()
+
+        # Add a colorbar for the filled contour.
+        fig.colorbar(im)
+
+        # Save the figure
+        print(" Saving figure as", outfile, "\n")
```

Comment:
I got feedback during the academies that it would be nice if all the plotting scripts printed out where the figure was being saved. I see this being done here, but just wanted to check if the other plotting scripts are doing the same. Thanks!

Reply:
@srherbener thanks for letting me know, I've added in 6ff7fae